### PR TITLE
Set up unit test environment for backend

### LINKED ISSUE #54: Setup backend test framework
Add a simple "hello world" type test, likely with the Jest framework.

### PR DESCRIPTION
**Pull request by jyrge**
Closes #54

### CODE REVIEW

File: server/src/index.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@ app.get('*', (req: Request, res: Response) => {
   res.send(`Hello ${req.path}`);
 });
 
-app.listen(port, () => {
-  console.log(`Hello server, running on port ${port}`);
-});
+if (require.main === module)
```
Comment (no-comma) [inline server/src/index.ts:15]: I would add brackets around this if statement since it's multiline.

File: server/package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
   "scripts": {
     "build": "npx tsc",
     "start": "node build/index.js",
+    "test": "jest",
```
Comment (no-comma) [inline server/package.json:11]: A note in the README about running tests with `npm test` would be good.

File: server/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/** @type {import('ts-jest').JestConfigWithTsJest} */
```
Comment (no-comma) [inline server/jest.config.js:1]: Why no REUSE header in this file?

Comment (jyrge) [reply server/jest.config.js:1]: I thought it as a config file

File: server/package.json (hunk 2 above)
Comment (jyrge) [reply server/package.json:11]: Agreed.

File: server/src/index.ts (hunk 1 above)
Comment (jyrge) [reply server/src/index.ts:15]: I have usually omitted curly braces from if statements if there is only one statement after it. Though in this case, it indeed may improve readability.

**Review by no-comma — APPROVED** [review]
LGTM

**Review by jj-stigell — APPROVED** [review]
Good!

**Pull request merged**